### PR TITLE
version management and documentation

### PR DESCRIPTION
**Pull request by yifan**
Use bump2version for version management

Use readthedocs.io for documentation management

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+# import sys
+# sys.path.insert(0, '/Users/yifan/QCRI/neurox/NeuroX/neurox')
+
+
+# -- Project information -----------------------------------------------------
+
+project = 'NeuroX toolkit'
+copyright = '2022, Qatar Computing Research Institute, HBKU'
+author = 'Qatar Computing Research Institute, HBKU'
+release = '1.0.9'
```
Comment (fdalvi) [inline docs/conf.py:23]: Should we read this from bump2version somehow to automate for the new version?

**Review by fdalvi — COMMENTED** [review]
Looks good @yifan, thanks a lot for working on this. Just one small change requested.

Apart from that, do you already have a test readthedocs page? I think it basically pulls the repo and builds the docs automatically yea? How is the versioning handled.

Finally, is it possible to build historic versions somehow? particularly it would be good to have docs from 1.0.6+ atleast!

File: docs/conf.py (hunk 1 above)
Comment (yifan) [reply docs/conf.py:23]: Apparently I missed this one :) good catch

**Comment by yifan** [comment]
> 

versions should be created automatically when we tag the release (bump2version should do it for you). 

For older version, we may need to do some manual work to build the docs. I hope adding tags is enough, if not, we need to create branch and add .readthedocs.yml to those branches, and then tag it.

**Pull request merged**